src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs

```diff
@@ -29,19 +41,28 @@ private static void PreloadWindowsLibrary(string dllFileName)
         {
             string basePath = Path.GetDirectoryName(typeof(Interop).Assembly.Location);
             string architecture = IntPtr.Size == 8 ? "x64" : "x86";
-            string dllPath = Path.Combine(basePath, architecture, dllFileName);
+            string dllPath = Path.Combine(basePath, architecture, $"{dllFileName}.dll");
 
             // return value is intentionally ignored as we let the subsequent P/Invokes fail naturally.
             LoadLibraryExW(dllPath, IntPtr.Zero, LOAD_WITH_ALTERED_SEARCH_PATH);
         }
 
+#if NETCOREAPP
+        private static IntPtr HostFxrDllImportResolver(string libraryName, Assembly assembly, DllImportSearchPath? searchPath)
+        {
+            if (libraryName != Constants.HostFxr || !NativeLibrary.TryLoad(HostFxrPath, out IntPtr handle))
+                return IntPtr.Zero;
```

Comment:
Same here - if it's possible to have some diagnostics... I think it would be better to fail with clear error if we fail to load the library (for whatever reason).